handler/dataprep/create.go

```diff
@@ -200,6 +265,187 @@ func (DefaultHandler) CreatePreparationHandler(
+	var wallets []model.Wallet
+	err := db.WithContext(ctx).
+		Joins("JOIN wallet_assignments ON wallets.id = wallet_assignments.wallet_id").
+		Where("wallet_assignments.preparation_id = ?", preparation.ID).
+		Find(&wallets).Error
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	if len(wallets) == 0 {
+		*validationErrors = append(*validationErrors, "No wallets assigned to preparation")
+
+		_, err = notificationHandler.LogWarning(ctx, db, "dataprep-create",
+			"No Wallets Found",
+			"No wallets are assigned to this preparation for auto-deal creation",
+			model.ConfigMap{
+				"preparation_name": preparation.Name,
+			})
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		return nil
+	}
+
+	// TODO: In a real implementation, you would connect to Lotus and validate each wallet
+	// For now, we'll just log that wallet validation is enabled
+	walletAddresses := make([]string, len(wallets))
+	for i, wallet := range wallets {
+		walletAddresses[i] = wallet.Address
+	}
+
+	_, err = notificationHandler.LogInfo(ctx, db, "dataprep-create",
+		"Wallet Validation Enabled",
+		"Wallet validation is enabled for auto-deal creation",
+		model.ConfigMap{
+			"preparation_name": preparation.Name,
+			"wallet_addresses": strings.Join(walletAddresses, ", "),
+		})
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	return nil
+}
+
+// performSPValidation validates storage provider for auto-deal creation
+func performSPValidation(ctx context.Context, db *gorm.DB, preparation *model.Preparation, validationErrors *[]string) error {
+	notificationHandler := notification.Default
+	spValidator := storage.DefaultSPValidator
+
+	// Check if a storage provider is specified
+	if preparation.DealConfig.DealProvider == "" {
+		// Try to get a default storage provider
+		defaultSP, err := spValidator.GetDefaultStorageProvider(ctx, db, "auto-deal-creation")
+		if err != nil {
+			*validationErrors = append(*validationErrors, "No storage provider specified and no default available")
+
+			_, err = notificationHandler.LogWarning(ctx, db, "dataprep-create",
+				"No Storage Provider Available",
+				"No storage provider specified and no default providers available",
+				model.ConfigMap{
+					"preparation_name": preparation.Name,
+				})
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			return nil
+		}
+
+		// Update preparation with default provider
+		preparation.DealConfig.DealProvider = defaultSP.ProviderID
+
+		_, err = notificationHandler.LogInfo(ctx, db, "dataprep-create",
+			"Default Storage Provider Selected",
+			"Using default storage provider for auto-deal creation",
+			model.ConfigMap{
+				"preparation_name": preparation.Name,
+				"provider_id":      defaultSP.ProviderID,
+				"provider_name":    defaultSP.Name,
+			})
+		if err != nil {
+			return errors.WithStack(err)
+		}
+	}
+
+	// TODO: In a real implementation, you would connect to Lotus and validate the storage provider
```

Comment:
I see this "in a real implementation" bit in a few places. These all seem like good suggestions -- are these part of the roadmap @Sankara-Jefferson @anjor ?